Composer/packages/client/src/pages/notifications/types.ts

```diff
@@ -1,13 +1,89 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+import { Diagnostic, createSingleMessage, DialogInfo, LuFile } from '@bfc/indexers';
+
+import { replaceDialogDiagnosticLabel } from '../../utils';
+
+export const DiagnosticSeverity = ['Error', 'Warning']; //'Information', 'Hint'
+
+export enum NotificationType {
+  DIALOG,
+  LG,
+  LU,
+}
 
 export interface INotification {
   id: string;
   severity: string;
-  type: string;
+  type: NotificationType;
   location: string;
   message: string;
   diagnostic: any;
+  dialogPath?: string; //the data path in dialog
 }
 
-export const DiagnosticSeverity = ['Error', 'Warning']; //'Information', 'Hint'
+export class DialogNotification implements INotification {
+  id: string;
+  severity: string;
+  type: NotificationType;
+  location: string;
+  message: string;
+  diagnostic: Diagnostic;
+  dialogPath?: string;
+  constructor(id: string, location: string, diagnostic: Diagnostic) {
+    this.id = id;
+    this.severity = DiagnosticSeverity[diagnostic.severity] || '';
+    this.message = `In ${replaceDialogDiagnosticLabel(diagnostic.path)} ${diagnostic.message}`;
+    this.diagnostic = diagnostic;
+    this.location = location;
+    this.type = NotificationType.DIALOG;
+    this.dialogPath = diagnostic.path;
+  }
+}
```

Comment:
Seems you need to define a base class `Notification` which implements `INotification` here, and use inheritance to build `Dialog/Lg/Lu Notification`.